app/src/main/java/com/internship/move/feature/map/MapFragment.kt

```diff
@@ -1,47 +1,261 @@
+        binding.scooterMapView.onResume()
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(requireContext())
+        getLocation()
         initViews()
+        initButtons()
+        initObservers()
+    }
+
+    private fun initViews() {
+        binding.scooterMapView.getMapAsync(this)
+    }
+
+    private fun initObservers() {
+        mapViewModel.scooterList.observe(viewLifecycleOwner) { scooterList ->
+            if (scooterList.isNotEmpty())
+                scooterList.forEach { scooterItem ->
+                    scooterMap.addMarker(
+                        MarkerOptions().position(
+                            LatLng(
+                                scooterItem.location.coordinates[1],
+                                scooterItem.location.coordinates[0]
+                            )
+                        ).title(scooterItem._id)
+                            .icon(bitmapDescriptorFromVector(R.drawable.ic_map_pin))
+                    )
+                }
+        }
+
+        mapViewModel.currentLocation.observe(viewLifecycleOwner) { location ->
+            scooterMap.moveCamera(CameraUpdateFactory.newLatLng(location))
+            mapViewModel.getScooters(location.latitude.toFloat(), location.longitude.toFloat())
+        }
+    }
+
+    private fun bitmapDescriptorFromVector(vectorResId: Int): BitmapDescriptor {
+        val vectorDrawable = ContextCompat.getDrawable(requireContext(), vectorResId)
+        vectorDrawable!!.setBounds(0, 0, vectorDrawable.getIntrinsicWidth(), vectorDrawable.getIntrinsicHeight())
+        val bitmap = Bitmap.createBitmap(vectorDrawable.getIntrinsicWidth(), vectorDrawable.getIntrinsicHeight(), Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(bitmap)
+        vectorDrawable.draw(canvas)
+        return BitmapDescriptorFactory.fromBitmap(bitmap)
     }
 
-    private fun initListeners() {
-        binding.logoutBtn.setOnClickListener {
-            authViewModel.logOut()
-            findNavController().navigate(MapFragmentDirections.actionMapFragmentToNavigationIntro())
+    private fun initButtons() {
+        requireActivity().onBackPressedDispatcher.addCallback {
+            if (doubleBackPressed) {
+                authViewModel.logOut()
+                requireActivity().finish()
+            } else {
+                doubleBackPressed = true
+                Toast.makeText(requireContext(), getString(R.string.exit_app_button_text), Toast.LENGTH_SHORT).show()
+                Handler().postDelayed({
+                    doubleBackPressed = false
+                }, 3000L)
+            }
         }
+    }
 
-        binding.clearBtn.setOnClickListener {
-            onboardingViewModel.changeLogStatus(logValue = false)
-            authViewModel.logOut()
+    private fun initScooterInfo(markerTitle: String) {
+        mapViewModel.scooterList.value?.forEach { scooter ->
+            if (scooter._id == markerTitle) {
```

Comment:
The `_id` field can be renamed to `id`. You're using `@Json(name = "_id")` to tell Moshi how to map it. This allows you to name the fields however you need